pkg/api/api.go

```diff
@@ -196,13 +196,16 @@ func (a *API) specificNodesHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	// TODO: these nodes don't send base target value at the moment
+	// TODO: these nodes don't send a generator at the moment
 	stateHashEvent := entities.NewStateHashEvent(
 		statement.Node,
 		zeroTS,
 		statement.Version,
 		statement.Height,
 		statehash,
 		zeroBT,
+		statehash.BlockID,
```

Comment:
I think the `BlockID` emptiness check should be added before using the field.

Reply:
Here it is not a pointer